system/jlib/jlog.cpp

```diff
@@ -2324,6 +2324,77 @@ MODULE_EXIT()
     thePassAllFilter = NULL;
 }
 
+#ifdef _CONTAINERIZED
+
+#define LOGFIELDS_ATT "@fields"
+#define LOGMSGDETAIL_ATT "@detail"
+#define LOGMSGAUDIENCES_ATT "@audiences"
+#define LOGMSGCLASSES_ATT "@classes"
+#define USELOGQUEUE_ATT "@useLogQueue"
+#define LOGQUEUELEN_ATT "@queueLen"
+#define LOGQUEUEDROP_ATT "@queueDrop"
+#define USESYSLOG_ATT "@enableSysLog"
+
+#ifdef _DEBUG
+  #define DEFAULT_USELOGQUEUE false
+#else
+  #define DEFAULT_USELOGQUEUE true
+#endif
+
+#define DEFAULT_QUEUELEN 512
+#define DEFAULT_QUEUEDROP 32
+#define DEFAULT_USE_SYSLOG false
+
+void setupContainerizedLogMsgHandler()
+{
+
+    Owned<IPropertyTreeIterator> iter = queryComponentConfig().getElements("*[1]/logging");
```

Comment:
we should only have one logging: section (ideally schema would validate that).
Not sure why you're iterating xpath "*[1]/logging" ?
Would expect it to still be:
```
IPropertyTree *logConfig = queryComponentConfig().queryPropTree("logging");
```

Reply:
The logging element is a child of the variable named "component" element, ie esp, roxie, etc.
I believe IPropertyTree *logConfig = queryComponentConfig().queryPropTree("logging"); targets the toplevel element. 
Also worth noting, I believe queryPropTree performs a getElements as well

Reply:
There should only be 1 "logging:" section per component.
Each component has it's own config file.
queryComponentConfig() will return the component section from that config file.

So for example:
```
    version: 1.0
    dali:
      name: mydali
      storage:
        storageClass: ""
        storageSize: 1Gi
      logging:
         detail: 50
    global:
      imageVersion: "master-2020-02-10"
      singleNode: false
```

(added logging: as an example)

So queryComponentConfig() when running in daserver will return the "dali" section as the root tree returned (this is defined by the call to loadConfiguration()). You will not have access to other sections (e.g. global:) 

logging: would be a property under dali: and generated if present in values.yaml.